src/components/styled-text-input/styled-text-input.tsx

```diff
@@ -1,37 +1,45 @@
-import React, { FC, useState } from 'react';
-import { NativeSyntheticEvent, TextInput, TextInputFocusEventData, TextInputProps } from 'react-native';
+import React, { FC } from 'react';
+import { TextInput, TextInputProps, View } from 'react-native';
 
-import { emptyFn } from '../../config/general';
-import { StyledTextInputStyles } from './styled-text-input.styles';
+import { useColors } from '../../styles/use-colors';
+import { IconNameEnum } from '../icon/icon-name.enum';
+import { TouchableIcon } from '../icon/touchable-icon/touchable-icon';
+import { useStyledTextInputStyles } from './styled-text-input.styles';
 
-export const StyledTextInput: FC<Omit<TextInputProps, 'style'>> = ({
+interface Props extends Omit<TextInputProps, 'style'> {
+  isError?: boolean;
+  isAllowCleanButton?: boolean;
+}
+
+export const StyledTextInput: FC<Props> = ({
+  onChangeText,
+  isAllowCleanButton,
```

Comment:
I have 2 options of naming for such cases: 
1. `isShow<Element name>`  
2. `has<Element name>`  
Use what do you prefer.

Also, set default values for `isAllowCleanButton` & `isError`